src/clj/rems/api/catalogue.clj

```diff
@@ -3,18 +3,23 @@
             [rems.api.schema :refer :all]
             [rems.api.util] ; required for route :roles
             [rems.db.catalogue :as catalogue]
+            [rems.roles :as roles]
+            [rems.auth.util :refer [throw-forbidden]]
             [ring.util.http-response :refer :all]
-            [schema.core :as s]))
+            [schema.core :as s]
+            [rems.config :refer [env]]
+            [rems.api.util :refer [check-roles]]))
 
 (s/defschema GetCatalogueResponse
   [CatalogueItem])
 
 (def catalogue-api
   (context "/catalogue" []
     :tags ["catalogue"]
-
     (GET "/" []
       :summary "Get the catalogue of items for the UI (does not include archived items)"
-      :roles #{:logged-in}
       :return GetCatalogueResponse
-      (ok (catalogue/get-localized-catalogue-items {:archived false})))))
+      (if (or (:catalogue-is-public env)
+       (roles/has-roles? :logged-in))
+       (ok (catalogue/get-localized-catalogue-items {:archived false}))
+       (throw-forbidden)))))
```

Comment:
Note how the `roles/has-roles?` is inside the `or`. Additionally, `if` is a macro so it gets two-space indentation. Thus:

```
(if (or (:catalogue-is-public env)
        (roles/has-roles? :logged-in))
  (ok (catalogue/get-localized-catalogue-items {:archived false}))
  (throw-forbidden)))))
```